core/trino-main/src/main/java/io/trino/operator/join/BigintJoinHash.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.join;
+
+import io.trino.spi.Page;
+import io.trino.spi.PageBuilder;
+import org.openjdk.jol.info.ClassLayout;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+// This implementation assumes arrays used in the hash are always a power of 2
+public final class BigintJoinHash
```

Comment:
Is it any different from `JoinHash`? If not, I think we can avoid `BigintJoinHash` because `JoinHash` class is isolated, see: `io.trino.sql.gen.JoinCompiler#internalCompileLookupSourceFactory`